pecos/core/utils/matrix.hpp

```diff
@@ -665,6 +740,43 @@ namespace pecos {
         return do_dot_product(y, x);
     }
 
+
+    template<class IX, class VX, class IY, class VY>
+    float32_t do_dot_product(const sdvec_t<IX, VX>& x, const sdvec_t<IY, VY>& y) {
+        if(x.nr_touch > y.nr_touch) {
+            return do_dot_product(y, x);
+        }
+        float32_t ret = 0;
+        for(size_t s=0; s < x.nr_touch; s++) {
```

Comment:
consider using the same style of for loop `for (size_t s = 0; s < ...; s++)` in all the functions for consistency.